paasta_tools/chronos_tools.py

```diff
@@ -225,7 +226,17 @@ def get_env(self):
         return [{"name": key, "value": value} for key, value in original_env.iteritems()]
 
     def get_constraints(self):
-        return self.config_dict.get('constraints', [['pool', 'EQUALS', self.get_pool()]])
+        if 'constraints' in self.config_dict:
+            constraints = self.config_dict.get('constraints')
+        else:
+            constraints = self.config_dict.get('extra_constraints', [])
+            constraints.extend(self.get_deploy_constraints())
+            constraints.extend(self.get_pool_constraints())
+        assert isinstance(constraints, list)
+        return [[str(val) for val in constraint] for constraint in constraints]
```

Comment:
I'd rather you moved all this logic to the caller, and have separate ``get_constraints`` and ``get_extra_constraints`` functions.